ultravox/tools/ds_tool.py

```diff
@@ -31,6 +31,19 @@ def __post_init__(self):
         if self.audio_column_name is None:
             self.audio_column_name = f"{self.column_name}_audio"
 
+    def map_split(self, ds_split: datasets.Dataset, num_proc: int) -> datasets.Dataset:
+        def map_sample(sample):
+            text = sample[self.column_name]
+            text = text["text"] if isinstance(text, dict) else text
+            sample[self.audio_column_name] = tts_client.tts(text)
+            return sample
```

Comment:
nit: given that `self` is already present, there's no need to use a `nested` function here.